lib/msf/core/post/windows/mssql.rb

```diff
@@ -109,13 +109,27 @@ def check_sqlcmd
         # @param [String] query the query to execute
         # @param [String] instance the SQL instance to target
         # @param [String] server the SQL server to target
+        # @param [String] database the database to connect to upon successfully authenticating
+        # @param [String] username the username to authenticate as
+        # @param [String] password the password to authenticate with
         # @return [String] the result of query
-        def run_sql(query, instance = nil, server = '.')
+        def run_sql(query, instance = nil, server = '.', database = nil, username = nil, password = nil)
```

Comment:
```suggestion
        def run_sql(query, instance = nil, server = '.', database: nil, username: nil, password: nil)
```

https://thoughtbot.com/blog/ruby-2-keyword-arguments

Reply:
Should we also be converting the other arguments then such as `server`?

Reply:
To be clear is the benefit here that the user could now specify an optional argument by name instead of position?

Reply:
Changes implemented in 7b5822dfb612680dd42e68857b8f8ee09484ca85

Reply:
> Should we also be converting the other arguments then such as server?

That would be a backwards incompatible change; You'd have to verify the old API isn't being used before changing it.

> To be clear is the benefit here that the user could now specify an optional argument by name instead of position?

Examples here: https://thoughtbot.com/blog/ruby-2-keyword-arguments#keyword-arguments-vs-positional-arguments

For me, I like that kwargs allows us to add additional parameters in the future without breaking backwards compatibility